src/Parser/Warlock/Affliction/Modules/Features/Abilities.js

```diff
@@ -146,68 +130,75 @@ class Abilities extends CoreAbilities {
+        spell: SPELLS.DARK_PACT_TALENT,
+        category: Abilities.SPELL_CATEGORIES.DEFENSIVE,
+        cooldown: 60,
+        isOnGCD: false,
+        enabled: combatant.hasTalent(SPELLS.DARK_PACT_TALENT.id),
         castEfficiency: {
           suggestion: false,
           importance: ISSUE_IMPORTANCE.MINOR,
-          recommendedEfficiency: 0.15,
-          averageIssueEfficiency: 0.07,
-          majorIssueEfficiency: 0.01,
+          recommendedEfficiency: 0.33,
+          averageIssueEfficiency: 0.20,
+          majorIssueEfficiency: 0.10,
         },
       },
+
+      // Utility
       {
-        spell: SPELLS.SOULSTONE,
+        spell: SPELLS.BURNING_RUSH_TALENT,
         category: Abilities.SPELL_CATEGORIES.UTILITY,
-        cooldown: 600,
-        // TODO: shares cooldown with other combat rezzes, don't know how to calculate properly
+        enabled: combatant.hasTalent(SPELLS.BURNING_RUSH_TALENT.id),
         isOnGCD: true,
       },
       {
-        spell: SPELLS.SUMMON_DOOMGUARD_TALENTED,
+        spell: SPELLS.DRAIN_LIFE,
         category: Abilities.SPELL_CATEGORIES.UTILITY,
-        enabled: combatant.hasTalent(SPELLS.GRIMOIRE_OF_SUPREMACY_TALENT.id),
         isOnGCD: true,
       },
       {
-        spell: SPELLS.SUMMON_INFERNAL_TALENTED,
+        spell: SPELLS.MORTAL_COIL_TALENT,
         category: Abilities.SPELL_CATEGORIES.UTILITY,
-        enabled: combatant.hasTalent(SPELLS.GRIMOIRE_OF_SUPREMACY_TALENT.id),
+        cooldown: 45,
+        enabled: combatant.hasTalent(SPELLS.MORTAL_COIL_TALENT.id),
         isOnGCD: true,
       },
       {
-        spell: SPELLS.DEMONIC_GATEWAY_CAST,
+        spell: SPELLS.DEMONIC_CIRCLE_SUMMON,
         category: Abilities.SPELL_CATEGORIES.UTILITY,
-        cooldown: 10,
+        enabled: combatant.hasTalent(SPELLS.DEMONIC_CIRCLE_TALENT.id),
         isOnGCD: true,
+        cooldown: 10,
+        castEfficiency: {
+          suggestion: false,
+          importance: ISSUE_IMPORTANCE.MINOR,
+        },
       },
       {
-        spell: SPELLS.DARK_PACT_TALENT,
+        spell: SPELLS.DEMONIC_CIRCLE_TELEPORT,
         category: Abilities.SPELL_CATEGORIES.UTILITY,
-        cooldown: 60,
-        enabled: combatant.hasTalent(SPELLS.DARK_PACT_TALENT.id),
+        cooldown: 30,
+        enabled: combatant.hasTalent(SPELLS.DEMONIC_CIRCLE_TALENT.id),
+        isOnGCD: true,
         castEfficiency: {
           suggestion: false,
           importance: ISSUE_IMPORTANCE.MINOR,
-          recommendedEfficiency: 0.33,
-          averageIssueEfficiency: 0.20,
-          majorIssueEfficiency: 0.10,
+          recommendedEfficiency: 0.15,
+          averageIssueEfficiency: 0.07,
+          majorIssueEfficiency: 0.01,
         },
       },
       {
-        spell: SPELLS.MORTAL_COIL_TALENT,
+        spell: SPELLS.SOULSTONE,
         category: Abilities.SPELL_CATEGORIES.UTILITY,
-        cooldown: 45,
-        enabled: combatant.hasTalent(SPELLS.MORTAL_COIL_TALENT.id),
+        cooldown: 600,
```

Comment:
I would recommend simply not setting a cooldown. You can't get the info from other people using resses.